tests/unit/amo/components/TestUserProfile.js

```diff
@@ -0,0 +1,321 @@
+      .toHaveText('hamsterdance.com/');
+  });
+
+  it("omits homepage if the user doesn't have one set", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        homepage: null,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-homepage')).toHaveLength(0);
+  });
+
+  it("renders the user's account creation date", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        created: '2000-08-15T12:01:13Z',
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-user-since')).toHaveLength(1);
+    expect(root.find('.UserProfile-user-since').children())
+      .toHaveText('Aug 15, 2000');
+  });
+
+  it('renders LoadingText for account creation date while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-tofu' } });
+
+    expect(root.find('.UserProfile-user-since')).toHaveLength(1);
+    expect(root.find('.UserProfile-user-since').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's number of add-ons", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        num_addons_listed: 70,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-number-of-addons')).toHaveLength(1);
+    expect(root.find('.UserProfile-number-of-addons').children())
+      .toHaveText('70');
+  });
+
+  it('renders LoadingText for number of add-ons while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-tofu' } });
+
+    expect(root.find('.UserProfile-number-of-addons')).toHaveLength(1);
+    expect(root.find('.UserProfile-number-of-addons').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's average add-on rating", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        average_addon_rating: 4.1,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-rating-average')).toHaveLength(1);
+    expect(root.find('.UserProfile-rating-average').find(Rating))
+      .toHaveProp('rating', 4.1);
+  });
+
+  it('renders LoadingText for average add-on rating while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-tofu' } });
+
+    expect(root.find('.UserProfile-rating-average')).toHaveLength(1);
+    expect(root.find('.UserProfile-rating-average').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's biography", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        biography: 'Not even vegan!',
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-biography').render().html())
```

Comment:
You can remove the `.render()` part. You only need `.html()`.

Reply:
Whoops, right on.

Reply:
Actually, no, the `.html()` will return an object. I'll try a different matcher.

Reply:
I think it's because it's doing dangerouslySetInnerHTML...